distributed/diagnostics/cluster_dump.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+from typing import Any, Collection, Dict, Literal
+
+from distributed.diagnostics.plugin import SchedulerPlugin
+from distributed.scheduler import Scheduler
+
+
+class ClusterDump(SchedulerPlugin):
+    """Dumps cluster state prior to Scheduler shutdown
+
+    The Scheduler may shutdown in cases where it is in an error state,
+    or when it has been unexpectedly idle for long periods of time.
+    This plugin dumps the cluster state prior to Scheduler shutdown
+    for debugging purposes.
+    """
+
+    def __init__(
+        self,
+        scheduler: Scheduler,
+        url: str = None,
```

Comment:
```suggestion
        url: str,
```
I'd prefer to make `url` required. If you're running an ephemeral cluster, having the default be that you'd write to an ephemeral disk that's about to disappear would be sad. I expect most cases where you'd actually want to use a `ClusterDump` plugin would be long-running cluster on ephemeral hardware. It's a good reminder that you need to specify this, and not too inconvenient in the rare cases you're actually running locally.

Reply:
Good point. I've made url required and removed the logic for providing a url in the case where it formerly was not provided.